### ISSUE TITLE
Getting a warning

### ISSUE THREAD

**Issue by Stardustwings** [CollectionFS/Meteor-CollectionFS#688]
I just follow the guide to test uploading simply. 
But I get the warning that
"The provided value 'undefined' is not a valid enum value of interface XMLHttpRequestResponseType."
in "http-call-client.js:101"
Is that normal using CollectionFS?

**Comment by rhyslbw** [comment]
Would need a reproduction repo to look into this

**Comment by yetanotherpete** [comment]
I've started getting this warning too, even though the files appear to upload successfully. I'm using the S3 adapter to two stores with graphicsmagic for resizing. This is with Chrome version 43.0.2357.81 (64-bit)

I don't recall receiving this warning previously. I just ran the same test on Safari and didn't get a warning. Any chance this is a Chrome thing?

Give me a yell if you want to test my build. It's not on GitHub, but I'm sure we can arrange something.

**Comment by thomaslove** [comment]
I'm getting this issue as well. Files uploading successfully with FS.Filesystem but still getting the same error. Chrome as well

**Comment by tinchoz49** [comment]
Same here! :+1: 

The uploader is working excellent but i'm getting the same warning.
 
Chrome version: 43.0.2357.81 (64 bit)
Adapter: GridFS

**Comment by EvilJohnny** [comment]
Yup! For me everything seems to be working but I am also getting the warning. Seems to be only in Chrome... at least Firefox doesn't give me any warnings.
I am using cfs:s3 with cfs:filesystem

**Comment by MichelFloyd** [comment]
I just started seeing this error on Mac/Chrome as well (but files seem to be uploading ok). Error does not occur on Mac/FF

**Comment by kenken17** [comment]
Having same warning here. Is that what we should fix on our side?

**Comment by constantinmusca** [comment]
I have also seen the warning. Any update on the matter ?

**Comment by Ethaan** [comment]
Same here

**Comment by MichelFloyd** [comment]
Still seeing this problem :(

**Comment by mgoldberg524** [comment]
I've made an [example repo](https://github.com/mgoldberg524/cfs-http-call-client-error-test-repo) that produces the warning for me on Google Chrome 43.0.2357.130 if it could be of some help.

**Comment by rhyslbw** [comment]
Hey guys, I'm quite time poor here at the moment to even look into this, but if anyone can resolve the problem I can review and roll it in

**Comment by kodamirmo** [comment]
I had the same issue on chrome mac but when I built the project the error was gone!!

**Comment by clarkritchie** [comment]
We're seeing this, too.

**Comment by genyded** [comment]
Not sure if the same fix would apply here, or where to make it but https://github.com/aldeed/meteor-http-extras/pull/8 had the same error and that pull fixed it. Seems to only happen in chrome and in dev mode.

**Comment by obiwarn** [comment]
same problem here

**Comment by almogdesign** [comment]
I'm getting this issue as well I take it's not fixed

**Comment by miri-am** [comment]
same here

**Comment by minafarid** [comment]
same here

**Comment by ekuiter** [comment]
same here (Chrome / Mac with gridfs / dropbox)

**Comment by mbabauer** [comment]
I am also getting this error: Mac / Chrome / filesystem

**Comment by WanielDeiss** [comment]
same here:
Chrome 45.0.2454.93 (64-bit)
Mac 10.10.5
Meteor 1.1.0.3
cfs:filesystem 0.1.2 
cfs:standard-packages 0.5.9

**Comment by johngonzalez** [comment]
same here: 
Versión 45.0.2454.93 (64-bit)
Ubuntu 14.04
Meteor 1.2.0.1
cfs:gridfs@0.0.33
cfs:s3@0.1.3
cfs:standard-packages@0.5.9

**Comment by udisun** [comment]
This issue isn't fixed yet.
Still have it, the problem is that the name is not taken from the file although the file contains a valid name property.

for some reason it looses it

**Comment by Maay** [comment]
+1

**Comment by Maay** [comment]
@udisun did u manage to fix it?

**Comment by yetanotherpete** [comment]
I'm beginning to think this project is mostly dead. Nothing has been updated in months and issues like this remain unresolved. 

I've since moved my project to [slingshot](https://github.com/CulturalMe/meteor-slingshot) and have had far fewer issues. It was a lot easier to configure too.

**Comment by udisun** [comment]
My pull request fixed the issue. But without the maintainer merging it we are stuck

**Comment by Maay** [comment]
@udisun i've made your changes locally but still got the same problem.

**Comment by thearabbit** [comment]
I have the same problem.
```js
cfs:standard-packages
cfs:gridfs
yogiben:autoform-file
```

**Comment by udisun** [comment]
@yetanotherpete do you still have the warning?
what about the actual file being saved to the local storage and the name() function, both of them should now show the name that was saved.
how can i get your changes and test for myself again?

**Comment by juanmaia** [comment]
same here
```
cfs:standard-packages
cfs:gridfs
yogiben:autoform-file
```
upload not working

**Comment by huvber** [comment]
same problem too. The problem for me is using it with ~~yogiben:autoform-modals~~

**Comment by veslo1** [comment]
+1

**Comment by Ajaxsoap** [comment]
Same here, for me when importing csv file, no error on the console and on the browser console it throws this error.

**Comment by Kostanos** [comment]
Same error here any updates on this issue?

**Comment by projectn00b** [comment]
Any luck?

**Comment by maximebodereau** [comment]
Same Here

**Comment by ansonla3** [comment]
+1

**Comment by awelles** [comment]
+1

**Comment by lgertel** [comment]
++++

**Comment by Sojourneer** [comment]
+1

**Comment by tastemeru** [comment]
The same error.
But looking at this topic I think that better to look for another upload system.

**Comment by MagneticMule** [comment]
Yep, I have the same error message using either the GridFS or S3 adapters. Chrome Canary on OSX and Chrome on Linux too. Haven't tried it on Windows.

The general consensus here seems to be that this project has pretty much been abandoned. Indeed, looking at some of the documentation from this project would support this suspicion.

Any ideas about alternatives? I know there is Slingshot but I need a generic FS solution. Not just S3.

**Comment by maximebodereau** [comment]
Yes use **maodouio:autoform-file** instead of yogiben:autoform-file

**Comment by ddresch** [comment]
@maximebodereau tried the package you mention and it's dropping the same error message.

`The provided value 'undefined' is not a valid enum value of type XMLHttpRequestResponseType.`

Using this setup:
cfs:standard-packages
cfs:gridfs
yogiben:autoform-file or maodouio:autoform-file

**Comment by Maay** [comment]
@ddresch I've managed to fix the problem for myself by using froked and unmerged local packages, you can download them here and then add them to your `package` dir inside the app dir. Specify then package versions in `/app/.meteor/packages` so they match local.
http://nekaka.com/d/qQ0titAHxh
Btw, i'm not sure which of the forked packages solved the problem, but it works and fine by me :)

**Comment by ddresch** [comment]
@Maay cool thanks a lot! Will have a look at this ZIP package later and try to figure out which of the local packages are needed.

**Comment by rafaelcorreiapoli** [comment]
+1

**Comment by mzygmunt** [comment]
+1

**Comment by andrewfulrich** [comment]
Looks like it was already fixed and has been sitting in a pull request since July: https://github.com/aldeed/meteor-http-extras/pull/6/commits

Am I mistaken?

**Comment by jOkes7** [comment]
+1

**Comment by lgertel** [comment]
No news for this?

**Comment by genyded** [comment]
A PR with a fix was submitted (months ago) but none of the repo devs seem to be active any longer so it and this issue are just sitting there. That seems to happen often with Meteor repos, but hopefully things will change when Meteor 1.3 release and packages become NPM based...

**Comment by lgertel** [comment]
Which PR?
TY!

**Comment by sotarules** [comment]
+1

**Comment by josejuanisaac** [comment]
I made a working uploader at the end of october with 3 other people and I tried bulding it from scratch today. The project we made is still working (file uploads correctly and no "The provided value 'undefined' is not a valid enum value of type XMLHttpRequestResponseType." error) while the project I made today is throwing this error. 
I can only conclude for now that the first project was "correct".
I will try to keep working on it to see if I can spot the difference between the two. I will post here if I find anything. If anyone wants to lend me their eyes, the project is https://github.com/avinoz/ningaloo.

**Comment by alejandrocarlos** [comment]
I'm getting this error as well on Mac/Chrome

**Comment by Krabaul** [comment]
**Image uploading in chrome is working again for me after I manually changed the files from the open Pull request:**

1. Create `packages` folder in your project

2. open your terminal and change location to your new packages folder

3. run `git clone https://github.com/aldeed/meteor-http-extras.git`

4. inside packages/meteor-http-extras open "packages.js" and rename the name to `"aldeed:http-fix"`


5. open "http-extras-client.js" and on line 95 change it from
 
		xhr.responseType = options.responseType;
		
	to

		//  FIX: https://github.com/aldeed/meteor-http-extras/pull/6/files
    	// if is undefined use an empty string by default
    	xhr.responseType = options.responseType || "";
    	
 6. Add package to your project: `meteor add aldeed:http-fix`
		

	
		
**Source: **

[How to modify/edit a Meteor package?](http://stackoverflow.com/a/30864393)

[Ignored Pull Request "fix issue 688"](https://github.com/aldeed/meteor-http-extras/pull/6/files)

**Comment by branco91** [comment]
@Krabaul i tried this, works well locally but when deployed to the server it complains it cant find the modified package ..

**Comment by lukaswelinder** [comment]
Getting the same warning:
Windows 10
Chrome
Meteor@1.2.1
cfs:standard-packages@0.5.9
cfs:gridfs@0.0.3

However, uploading seems to work just fine despite the warning. Haven't deployed to test it yet.

As much as I like this package, I'm going to be looking for an alternative that's more likely to be kept up as Meteor moves into 1.3... Anyone have suggestions?

**Comment by tuanhai12h** [comment]
Dear team,

I know it can fix
` xhr.responseType = options.responseType || "";`

Why it so long to be fixed?
Please update it soon. Thanks

**Comment by obiwarn** [comment]
@tuanhai12h Did you file a PR?

**Comment by tuanhai12h** [comment]
@obiwarn same this https://github.com/CollectionFS/DEPRECATING-cfs-upload-http/blob/master/packages/upload-http/http-call-client.js at line 101.

I don't know why in the Code in HTTP.Call but the packages downloaded is used httpCall same the DEPRECATING-cfs-upload-http (link above)

**Comment by cesarve77** [comment]
+1

**Comment by jlukic** [comment]
@aldeed This is a low hanging fruit. I know its difficult to mantain so many packages, but this might be worth checking out when you have a free moment.

**Comment by S-C-Harish** [comment]
+1 I am facing the same error

**Comment by mngkvn** [comment]
+1

**Comment by polux2016** [comment]
+1

**Comment by nickgermaine** [comment]
Wow.  Almost a year after a pr was submitted and people still having this problem... smh.

This warning prevents the interface from updating (react/meteor) to display the newly uploaded image...

**Comment by wardloockx** [comment]
+1

**Comment by designandreev** [comment]
+1

**Comment by jmlv** [comment]
+1

**Comment by joncursi** [comment]
+1

**Comment by mngkvn** [comment]
https://forums.meteor.com/t/deprecating-collectionfs/16921

**Comment by lukaswelinder** [comment]
Thanks for the link @mngkvn

**Comment by adishua** [comment]
+1

**Comment by andhdo** [comment]
+1

**Comment by highgrav** [comment]
+1

**Comment by varunajmera0** [comment]
i replaced it
//  FIX: https://github.com/aldeed/meteor-http-extras/pull/6/files
// if is undefined use an empty string by default
xhr.responseType = options.responseType || "";

But whenever server restarts it replaces previous file.

**Comment by imike57** [comment]
+1

**Comment by tlongren** [comment]
+1 :unamused:

**Comment by micktaiwan** [comment]
+1

**Comment by iehdk** [comment]
+1
